### ISSUE TITLE
Document not inserting after upgrade to 4.4.10+

### ISSUE THREAD

**Issue by alocher** [Automattic/mongoose#5830]
Node: 8.9.1
Mongoose: 4.4.10

My code runs fine with mongoose 4.4.9, but when I upgrade to 4.4.10 and later, the code "breaks" at one specific spot and I can't seem to figure out why. I'm trying to insert (create) a document in my collection but the document never gets created and also doesn't throw an error. All other insert/updates are not affected.

The method in question is below and everything seems to run smoothly as I get to line 34 of that code. Afterwards if I check my database, the document hasn't been written. I'm not getting any error message from mongoose or mongodb.
https://github.com/tatool/tatool-web/blob/master/controllers/repositoryCtrl.js#L19 

Since everything works fine in 4.4.9 I hoped to see what changes in 4.4.10 might be responsible but wasn't able to pinpoint anything.

**Comment by superheri** [comment]
Why not upgrading to the latest version of mongoose?

**Comment by alocher** [comment]
Sorry I should have been more clear, this breaks starting from 4.4.10 up including to the latest version (4.13.4). I've just tried to narrow down the release when it started to break.

**Comment by vkarpov15** [comment]
Can you provide instructions on how I can reproduce this issue? Is there a test in your `tests` folder that can reproduce this issue?

**Comment by alocher** [comment]
will create a test for it and let you know.

**Comment by varunjayaraman** [comment]
hey @alocher any update on this?

**Comment by alocher** [comment]
I know this is late but I had to put this aside for a while. After picking it up again I found the issue. The problem was that I was copying a document from one collection to another. I was removing the _id to make sure this would be a new document. It turns out mongoose was always doing an update instead of an insert as it expected this was an existing document. I had to add the parameter doc.IsNew = true to force an insert.
